server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -215,8 +215,18 @@ public ClusterState execute(ClusterState currentState) {
                         final RepositoryMetadata updatedMetadata;
                         if (canUpdateInPlace(newRepositoryMetadata, existing)) {
                             if (repositoryMetadata.settings().equals(newRepositoryMetadata.settings())) {
-                                // Previous version is the same as this one no update is needed.
-                                return currentState;
+                                if (repositoryMetadata.generation() == RepositoryData.CORRUPTED_REPO_GEN) {
+                                    // If re-creating a repository with the same settings, reset the corrupted flag (issue #89130).
+                                    // Setting the safe generation to unknown, so that a consistent generation is found.
+                                    repositoryMetadata = new RepositoryMetadata(
+                                        repositoryMetadata,
+                                        RepositoryData.UNKNOWN_REPO_GEN,
+                                        repositoryMetadata.pendingGeneration()
+                                    );
```

Comment:
NIT: I wonder if it is worth adding `withGeneration` to make it clear what we are changing?

Reply:
Thanks @idegtiarenko ! Done